copy.c

```diff
@@ -1,11 +1,46 @@
 #include <stdio.h>
 #include <windows.h>
+#include <fcntl.h>
+
+typedef struct node_tag {
+	char buffer[1024];
+	int capacity;
+	struct node_tag *next;
+} node;
+
+int wmain() {
+	_setmode(_fileno(stdin), _O_U8TEXT);
+
+	node* bufferList = malloc(sizeof(node));
+	bufferList->capacity = 0;
+	node* head = bufferList;
+
+	DWORD cbSize = 0;
+	DWORD count;
+	while ((count = fread(head->buffer, 1, sizeof(head->buffer), stdin)) != 0 && !ferror(stdin)) {
+		cbSize += count;
+		head->capacity = count;
+		head->next = malloc(sizeof(node));
```

Comment:
I'm in the camp of always freeing your resources, even if the system is going to clean them up for you.

Also, `NULL` check is imperative because it signals that the system is out of memory. The least you can do is return gracefully from it instead of causing an access violation.